Patch by Brendan Tildesley <mail@brendan.scot>

Fixes 2 non-reproducibility bugs:
1: root GID in guix can changed, and is embedded in the tarball. this sets it to 0
2: mtime of some directories set to current time. This sets it to 0 SOURCE_DATE_EPOCH

See https://bugs.kde.org/show_bug.cgi?id=443532
--- a/kde-modules/KDEPackageAppTemplates.cmake
+++ b/kde-modules/KDEPackageAppTemplates.cmake
@@ -117,8 +117,8 @@ function(kde_package_app_templates)
             # Make tar archive reproducible, the arguments are only available with GNU tar
             add_custom_command(OUTPUT ${_template}
                 COMMAND ${_tar_executable} ARGS -c
-                   --exclude .kdev_ignore --exclude .svn --sort=name --mode=go=rX,u+rw,a-s --owner=root
-                   --pax-option=exthdr.name=%d/PaxHeaders/%f,atime:=0,ctime:=0
-                   --group=root --numeric-owner -j -v -f ${_template} .
+                   --exclude .kdev_ignore --exclude .svn --sort=name --mode=go=rX,u+rw,a-s
+                   --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime
+                   --mtime="@${SOURCE_DATE_EPOCH}" --owner=0 --group=0 --numeric-owner -j -v -f ${_template} .
                 WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/${_templateName}
                 DEPENDS ${_subdirs_entries}
             )
